File: cpp/001.Two Sum.cpp
class Solution{
public:
	vector<int> twoSum(vector<int>& nums, int target){
		unordered_map<int, int> mp;
		vector<int> result;
		for(int i = 0; i < nums.size(); i++){
			if(mp.find(target - nums[i]) == mp.end()){
				mp[nums[i]] = i;
			}else{
				result.push_back(mp[target - nums[i]]);
				result.push_back(i);
			}
		}

		return result;
	}
};

File: Java/001.Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> mp = new HashMap<>();
        int[] result = new int[2];

        for(int i = 0 ; i < nums.length; i++){
        	if(mp.containsKey(target - nums[i])){
        		result = new int[] {mp.get(target - nums[i]), i};
        		break;
        	}
        	else{
        		mp.put(nums[i], i);
        	}
        }

        return result;
    }
}